lib/MiLight/RadioStack.h

```diff
@@ -2,33 +2,55 @@
 #include <PL1167_nRF24.h>
 #include <MiLightRadioConfig.h>
 #include <MiLightRadio.h>
+#include <MiLightRadioPL1167_LT8900.h>
+#include <MiLightRadioInterface.h>
+#include <Settings.h>
 
 #ifndef _RADIO_STACK_H
-#define _RADIO_STACK_H 
+#define _RADIO_STACK_H
 
 class RadioStack {
```

Comment:
I'm thinking that `RadioStack` was and is kind of serving the role of a poorly named and half-heartedly implemented Factory. Its original purpose was to contain the logic to construct an instance of `MiLightRadio`, but it's not doing a very good job of that since you still have to pass in an instance of `RF24`.

I think if we renamed this `RadioProviderFactory` and had two impls that took in only the radio config and the hardware config (pins, etc.), we could just pass that into `MiLightClient` and the separation would be a whole lot cleaner.